### PR TITLE
Other yaml changes are not integrated when valid

### PR DESCRIPTION
**Pull request by RobDolares**
related Issues:
https://issues.redhat.com/browse/ACM-4449
https://issues.redhat.com/browse/ACM-4617

### CODE REVIEW

File: frontend/public/locales/en/translation.json [hunk 1]
```diff
@@ -27,6 +27,8 @@
   "{{remaining}} more": "{{remaining}} more",
   "{{remoteCount}} Remote": "{{remoteCount}} Remote",
   "{{remoteCount}} Remote, 1 Local": "{{remoteCount}} Remote, 1 Local",
+  "{{testInventory}} is not an existing ansible inventory": "{{testInventory}} is not an existing ansible inventory",
+  "{{testSecret}} is not an existing ansible credential": "{{testSecret}} is not an existing ansible credential",
```
Comment (KevinFCormier) [inline frontend/public/locales/en/translation.json:31]: 1. Ansible should always start with capital A.
2. Why are interpolation variables called `testX`? `{{inventory}}` and `{{credential}}` would be the natural choices.
3. Why is the name of the credential having the first letter uppercased? (See screenshot.)

image

File: frontend/src/components/SyncEditor/SyncEditor.test.tsx [hunk 2]
```diff
@@ -292,6 +293,16 @@ describe('SyncEditor component', () => {
...
+      {
+        path: `ClusterCurator[0].spec.install.jobMonitorTimeout}`,
+        setter: ((value: any) => {
+          const errors: any[] = []
+          const path = `ClusterCurator[0].spec.prehook`
+          errors.push({ path, message: 'Must be Job or Workflow' })
+          setInstallTimeout(value)
+          return errors
+        }).bind(null),
+      },
```
Comment (KevinFCormier) [inline frontend/src/components/SyncEditor/SyncEditor.test.tsx:305]: This branch may be out of date... was the work started before we pulled the timeout story?

File: frontend/src/routes/Infrastructure/Automations/AnsibleAutomationsForm.tsx [hunk 3]
```diff
@@ -281,8 +282,96 @@ export function AnsibleAutomationsForm(props: {
     }
     return curator
   }
+
+  const jobTypes = [{ type: 'install' }, { type: 'upgrade' }, { type: 'scale' }, { type: 'destroy' }]
+
```
Comment (KevinFCormier) [inline frontend/src/routes/Infrastructure/Automations/AnsibleAutomationsForm.tsx:287]: We have a recoil selector to get the current supported Ansible job types. Probably no need to repeat them out here.

File: frontend/src/routes/Infrastructure/Automations/AnsibleAutomationsForm.tsx [hunk 4]
```diff
@@ -281,8 +282,96 @@ export function AnsibleAutomationsForm(props: {
...
+    return errors
+  }
+
+  function setCustomHook(type: string, preHook: boolean, value: any) {
+    const errors: any[] = []
+    if (Array.isArray(value)) {
+      value.forEach(({ type: jobType, name }, index) => {
+        const path = `ClusterCurator[0].spec.${type}.${preHook ? 'prehook' : 'posthook'}.${index}`
+        if (!['Job', 'Workflow'].includes(jobType)) {
+          errors.push({ path: `${path}.type`, message: 'Must be Job or Workflow' })
```
Comment (KevinFCormier) [inline frontend/src/routes/Infrastructure/Automations/AnsibleAutomationsForm.tsx:335]: Untranslated message

**Review by KevinFCormier — CHANGES_REQUESTED** [review]
Please see inline comments @jeswanke and/or @RobDolares

File: frontend/public/locales/en/translation.json (hunk 1 above)
Comment (jeswanke) [reply frontend/public/locales/en/translation.json:31]: done, done, done

File: frontend/src/routes/Infrastructure/Automations/AnsibleAutomationsForm.tsx (hunk 4 above)
Comment (jeswanke) [reply frontend/src/routes/Infrastructure/Automations/AnsibleAutomationsForm.tsx:335]: done

File: frontend/src/routes/Infrastructure/Automations/AnsibleAutomationsForm.tsx (hunk 3 above)
Comment (jeswanke) [reply frontend/src/routes/Infrastructure/Automations/AnsibleAutomationsForm.tsx]: cool, using now

File: frontend/src/components/SyncEditor/SyncEditor.test.tsx [hunk 5]
```diff
@@ -1319,7 +1330,7 @@ const semanticErrors = [
     options: {
       isWholeLine: true,
       glyphMarginClassName: 'errorDecoration',
-      glyphMarginHoverMessage: { value: "```html\nMust have required property 'name' \n```" },
+      glyphMarginHoverMessage: { value: "```html\nmust have required property 'name' \n```" },
```
Comment (KevinFCormier) [inline frontend/src/components/SyncEditor/SyncEditor.test.tsx:1333]: @jeswanke Why did you change these? I didn't have a problem with the error messages starting with a capital letter in general - it was just when you were using user input that did not match what the user had entered in the YAML editor.

File: frontend/public/locales/en/translation.json [hunk 6]
```diff
@@ -27,6 +28,8 @@
   "{{remaining}} more": "{{remaining}} more",
   "{{remoteCount}} Remote": "{{remoteCount}} Remote",
   "{{remoteCount}} Remote, 1 Local": "{{remoteCount}} Remote, 1 Local",
+  "{{testInventory}} is not an existing ansible inventory": "{{testInventory}} is not an existing Ansible inventory",
+  "{{testSecret}} is not an existing ansible credential": "{{testSecret}} is not an existing Ansible credential",
```
Comment (KevinFCormier) [inline frontend/public/locales/en/translation.json:32]: Might as well make the key match the value since it is a new message. The difference in case might be confusing.

File: frontend/src/routes/Infrastructure/Automations/AnsibleAutomationsForm.tsx [hunk 7]
```diff
@@ -281,8 +284,94 @@ export function AnsibleAutomationsForm(props: {
...
+    if (Array.isArray(value)) {
+      value.forEach(({ type: jobType, name }, index) => {
+        const path = `ClusterCurator[0].spec.${type}.${preHook ? 'prehook' : 'posthook'}.${index}`
+        if (!['Job', 'Workflow'].includes(jobType)) {
+          errors.push({ path: `${path}.type`, message: t('Must be Job or Workflow') })
+        }
+        if (name && !AnsibleTowerJobTemplateList?.includes(name)) {
+          errors.push({ path: `${path}.name`, message: t('"{{name}}" is not an existing job', { name }) })
+        }
+      })
```
Comment (KevinFCormier) [inline frontend/src/routes/Infrastructure/Automations/AnsibleAutomationsForm.tsx:340]: We need to check the `AnsibleTowerWorkflowTemplateList` if the type is `Workflow`.

File: frontend/public/locales/en/translation.json [hunk 8]
```diff
@@ -2,6 +2,7 @@
   " No status: ": " No status: ",
   " With violations: ": " With violations: ",
   " Without violations: ": " Without violations: ",
+  "\"{{name}}\" is not an existing job": "\"{{name}}\" is not an existing job",
```
Comment (KevinFCormier) [inline frontend/public/locales/en/translation.json:5]: This one does not include "Ansible" even though all the others do. Also, we should probably either have separate messages for "Ansible job templates" and "Ansible workflow job templates".

File: frontend/public/locales/en/translation.json [hunk 9]
```diff
@@ -2,6 +2,9 @@
   " No status: ": " No status: ",
   " With violations: ": " With violations: ",
   " Without violations: ": " Without violations: ",
+  "\"{{name}}\" is not an Ansible existing job": "\"{{name}}\" is not an Ansible existing job",
```
Comment (KevinFCormier) [inline frontend/public/locales/en/translation.json:5]: Slow down, John 😛

Comment (jeswanke) [reply frontend/public/locales/en/translation.json:5]: ha ;)

**Review by KevinFCormier — APPROVED** [review]
